src/core/lib/security/transport/security_handshaker.cc

```diff
@@ -403,7 +403,22 @@ static void security_handshaker_do_handshake(grpc_exec_ctx *exec_ctx,
   h->args = args;
   h->on_handshake_done = on_handshake_done;
   gpr_ref(&h->refs);
-  grpc_error *error = do_handshaker_next_locked(exec_ctx, h, NULL, 0);
+  size_t bytes_received_size = args->read_buffer->length;
```

Comment:
This code looks almost identical to the existing code in `on_handshake_data_received_from_peer()`.  How about refactoring to avoid duplication?